python/tests/server/test_worker.py

```diff
@@ -103,16 +115,22 @@ class Result:
     exception: Optional[Exception] = None
 
 
-def _process(events, swallow_exceptions=False):
+async def make_generator_async(it: Iterator[T]) -> AsyncIterator[T]:
+    for item in iter(it):
+        yield item
+
+
+async def _process(events, swallow_exceptions=False):
     """
     Helper function to collect events generated by Worker during tests.
     """
     result = Result()
     stdout = []
     stderr = []
-
+    if not inspect.isasyncgen(events):
```

Comment:
I couldn't find anywhere that `_process` is called with a non-async generator. Do we need this and `make_generator_async`?

Reply:
Oh, I found it, in the Hypothesis test. Seems to me like it would be simpler and less surprising to define an `_aprocess` and use that where appropriate:

```python
async def _aprocess(events, swallow_exceptions=False):
    events = []
    async for event in events:
        events.append(event)
    return _process(events, swallow_exceptions=swallow_exceptions)
```

Reply:
something like that was the first thing I tried, that won't catch exceptions correctly

Reply:
specifically, if you do it that way, swallow_exceptions will not be respected; if you instead put the try/catch block around the `[e async for e in events]`, then you miss the events leading up to the exception

I would consider moving make_generator_async to the part of hypothesis that passes in a list though

Reply:
oh that would have been extremely helpful to know about swallow_exceptions

I've pushed an attempt to be less surprising that keeps the most common async usecase easy, but I can do it your way if you feel adamant